server/implementation/src/main/java/io/smallrye/graphql/execution/event/EventEmitter.java

```diff
@@ -0,0 +1,196 @@
+            EventEmitter eventEmitter = new EventEmitter(config);
+            current.set(eventEmitter);
+        }
+    }
+
+    public static void end() {
+        current.remove();
+    }
+
+    // While building the schema
+
+    /**
+     * This gets fired just before we create the final schema. This allows listeners to add to the builder any
+     * custom elements.
+     * 
+     * @param builder as it stands
+     * @return builder modified by listener
+     */
+    public static GraphQLSchema.Builder fireBeforeSchemaBuild(GraphQLSchema.Builder builder) {
+        Optional<EventEmitter> emitter = emitter();
+        if (emitter.isPresent()) {
+            return emitter.get().emmitBeforeSchemaBuild(builder);
+        }
+        return builder;
+    }
+
+    /**
+     * This gets fired during the bootstrap phase before a new operation
+     * is being created. This allows listeners to modify the operation
+     * 
+     * @param operation as it stands
+     * @return operation possibly modified
+     */
+    public static Operation fireCreateOperation(Operation operation) {
+        Optional<EventEmitter> emitter = emitter();
+        if (emitter.isPresent()) {
+            return emitter.get().emmitCreateOperation(operation);
+        }
+        return operation;
+    }
+
+    // While execuring requests
+
+    public static void fireBeforeExecute() {
+        emitter().ifPresent((t) -> t.emmitBeforeExecute());
+    }
+
+    public static void fireOnExecuteError(String executionId, Throwable t) {
+        Optional<EventEmitter> emitter = emitter();
+        if (emitter.isPresent()) {
+            emitter.get().emmitOnExecuteError(executionId, t);
+        }
+    }
+
+    public static void fireAfterExecute() {
+        emitter().ifPresent((t) -> t.emmitAfterExecute());
+    }
+
+    public static void fireBeforeDataFetch() {
+        emitter().ifPresent((t) -> t.emmitBeforeDataFetch());
+    }
+
+    public static void fireBeforeMethodInvoke(InvokeInfo invokeInfo) throws Exception {
+        Optional<EventEmitter> emitter = emitter();
+        if (emitter.isPresent()) {
+            emitter.get().emmitBeforeDataFetchMethodInvoke(invokeInfo);
+        }
+    }
+
+    public static void fireOnDataFetchError(String executionId, Throwable t) {
+        Optional<EventEmitter> emitter = emitter();
+        if (emitter.isPresent()) {
+            emitter.get().emmitOnDataFetchError(executionId, t);
+        }
+    }
+
+    public static void fireAfterDataFetch() {
+        emitter().ifPresent((t) -> t.emmitAfterDataFetch());
+    }
+
+    private static Optional<EventEmitter> emitter() {
+        EventEmitter eventEmitter = current.get();
+        if (eventEmitter != null) {
+            return Optional.of(eventEmitter);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private static final ServiceLoader<EventingService> eventingServices = ServiceLoader.load(EventingService.class);
```

Comment:
This must not be static. In WildFly, the `EventEmitter` class will be loaded by the same class loader for all deployments, therefore the value of static field in it would be shared between all deployments, which is wrong.

In general, to work with WildFly, we need to avoid static fields as much as possible.